src/TorchSharp/JIT/ScriptModule.cs

```diff
@@ -0,0 +1,333 @@
+                }
+
+                [DllImport("LibTorchSharp")]
+                private static extern long THSJIT_getNumModules(HType module);
+
+                [DllImport("LibTorchSharp")]
+                private static extern int THSJIT_Module_num_inputs(HType module);
+
+                public int GetNumberOfInputs()
+                {
+                    return THSJIT_Module_num_inputs(handle);
+                }
+
+                [DllImport("LibTorchSharp")]
+                private static extern int THSJIT_Module_num_outputs(HType module);
+
+                public int GetNumberOfOutputs()
+                {
+                    return THSJIT_Module_num_outputs(handle);
+                }
+
+                [DllImport("LibTorchSharp")]
+                private static extern void THSJIT_Module_train(HType module);
+
+                public override void train()
+                {
+                    THSJIT_Module_train(handle);
+                    CheckForErrors();
+                }
+
+                [DllImport("LibTorchSharp")]
+                private static extern void THSJIT_Module_eval(HType module);
+
+                public override void eval()
+                {
+                    THSJIT_Module_eval(handle);
+                    CheckForErrors();
+                }
+
+                [DllImport("LibTorchSharp")]
+                private static extern bool THSJIT_Module_is_training(HType module);
+
+                public override bool training {
+                    get {
+                        var res = THSJIT_Module_is_training(handle);
+                        CheckForErrors();
+                        return res;
+                    }
+                }
+
+                [DllImport("LibTorchSharp")]
+                static extern void THSJIT_Module_to_device(HType module, long deviceType, long deviceIndex);
+
+                [DllImport("LibTorchSharp")]
+                static extern void THSJIT_Module_to_dtype(HType module, sbyte dtype);
+
+                /// <summary>
+                /// Moves the parameters and buffers.
+                /// </summary>
+                /// <param name="deviceType">The device type, e.g. 'CPU' or 'CUDA'.</param>
+                /// <param name="deviceIndex">The optional device index.</param>
+                /// <returns></returns>
+                public override nn.Module to(DeviceType deviceType, int deviceIndex = -1)
+                {
+                    if (deviceType != DeviceType.CUDA) deviceIndex = -1;
+
+                    if (deviceType == DeviceType.CUDA && !torch.cuda.is_available()) throw new InvalidOperationException("CUDA is not available.");
+
+                    if (deviceType != _deviceType || deviceIndex != _deviceIndex) {
+
+                        InitializeDeviceType(deviceType);
+                        THSJIT_Module_to_device(handle, (int)deviceType, deviceIndex);
```

Comment:
>deviceIndex [](http://example.com/codeflow?start=73&length=11)

for my education, what is the behavior when calling THSJIT_Module_to_device and passing the deviceIndex = -1?

Reply:
It picks the default GPU.